### ISSUE TITLE
Mouse clicks need to be done twice before it registers on the webpage/form

### ISSUE THREAD

**Issue by RupertAvery** [chromelyapps/Chromely#93]
- Using the latest code from master branch.
- Using the Vue demo app.
- Directly referencing Chromely.CefSharp.WinApi and Chromely.Core projects in my project.

Creating the window like so:

```csharp
                ChromelyConfiguration config = ChromelyConfiguration
                                              .Create(container)
                                              .WithAppArgs(args)
                                              .WithHostSize(1200, 900)
                                              .WithHostMode(WindowState.Normal, true)
                                              .WithHostTitle("delphi")
                                              .WithHostIconFile("chromely.ico")
                                              .WithStartUrl(startUrl)
                                              .WithLogFile("logs\\chromely.cef_new.log")
                                              .WithLogSeverity(LogSeverity.Info)
                                              .UseDefaultLogger()
                                              .UseDefaultResourceSchemeHandler("local", string.Empty)
                                              .UseDefaultHttpSchemeHandler("http", "chromely.com")
                                              .UseDefaultJsHandler("boundControllerAsync", true);


                using (var window = ChromelyWindow.Create(config))
                {
                    window.RegisterUrlScheme(new UrlScheme("https://github.com/mattkol/Chromely", true));
                    // 1. Register current/local assembly:
                    window.RegisterServiceAssembly(Assembly.GetExecutingAssembly());
                    window.ScanAssemblies();

                    return window.Run(args);
                }
```

For some reason I need to click twice to interact with anything on the window. Clicking on the page causes the window to be draggable.

**Comment by RupertAvery** [comment]
Clicks aren't evident in this screencap, but I'm clicking twice on the minimize/maximze/close buttons.

![chromely](https://user-images.githubusercontent.com/1910659/57529505-aa730080-7367-11e9-9730-4c3d2dda31d7.gif)

**Comment by mattkol** [comment]
@RupertAvery this is due to a recent merge of PR #84 to master branch. 

Please try get latest and see if the issue is still there.

Thanks for pointing this out.

@NovusTheory @FrankPfattheicher please can you help review the changes I made to PR #84 merge - https://github.com/chromelyapps/Chromely/commit/44f8abee7053e3b255e3dda8b040bb23fd07ee47

We can start a new issue to track this is if need be. Please advise.

@NovusTheory the draggable handler is not required. And if it is needed, it should be registered like:

````
.RegisterCustomHandler(CefHandlerKey.DragHandler, typeof(CustomCefSharpDragHandler))
````
@NovusTheory @FrankPfattheicher great job on frameless and draggable!!! Cool feature!

**Comment by NovusTheory** [comment]
In all my testing I've never had a double click issue. And yes the drag handler is needed as it registers the webkit-app-region

**Comment by mattkol** [comment]
@NovusTheory  @FrankPfattheicher there was an issue for real. I could not even get to run when I tested it. But we should be in better shape now. I only let the changes impact frameless option. As we progress with the implementation we can gradually bring the functionality to the rest of the process. These changes can take sometime to get right for all options.

Also the drag worked great in CefGlue, but not for CefSharp .. I had to sync the  WndProcOverride class/implementation - please help check that I am not breaking anything.

So the drag handler is needed. I will suggest we only apply it to frameless for now and as we learn more of the impact we can apply to all. So I will bring the drag handler back, but only when frameless is selected as a default drag handler.

I also noticed there is no drag handler for CefGlue. It does not need a drag handler or we need to do what is done in CefSharp to CefGlue?

**Comment by NovusTheory** [comment]
@mattkol The drag handler was not added in CefGlue as the System.Drawing class doesn't exist. The implementation isn't considered complete for CefGlue. If an alternative can be used to add the necessary classes then it can be done

**Comment by RupertAvery** [comment]
This seems to have been fixed by commit 44f8abee7053e3b255e3dda8b040bb23fd07ee47, thanks!
